src/client/conn.rs

```diff
@@ -223,39 +224,82 @@ impl Connection {
         config: &Config,
         tx: UnboundedSender<Message>,
     ) -> error::Result<Transport<TlsStream<TcpStream>>> {
-        struct DangerousAcceptAllVerifier;
+        #[derive(Debug)]
+        struct DangerousAcceptAllVerifier(CryptoProvider);
+
+        impl DangerousAcceptAllVerifier {
+            fn new() -> Self {
+                DangerousAcceptAllVerifier(CryptoProvider::get_default().unwrap().as_ref().clone())
```

Comment:
I think life will be easier in this case if you have `DangerousAcceptAllVerifier` hold `Arc<CryptoProvider>` instead of `CryptoProvider`. Afterwards this can boil down to just `CryptoProvider::get_default().unwrap()`.

Or, for maximal debuggability, `CryptoProvider::get_default().expect("no process default crypto provider has been set - application must call CryptoProvider::install_default()")`